src/main/java/service/impl/DataConverterImpl.java

```
@@ -0,0 +1,31 @@
+package core.basesyntax;
+
+import core.interfaces.DataConverter;
+import java.util.Arrays;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        return data.stream()
+                .skip(1)
+                .filter(s -> !s.isEmpty())
+                .map(this::toTransaction)
+                .toList();
+    }
+
+    private FruitTransaction toTransaction(String data) {
+        String[] split = data.split(",");
+        if (split.length != 3) {
+            throw new IllegalArgumentException("Bad line" + Arrays.toString(split));
+        }
+        FruitTransaction.Operation operation
+                = FruitTransaction.Operation.fromCode(split[0]);
+        String fruit
+                = split[1];
+        int quantity
+                = Integer.parseInt(split[2]);
+        return new FruitTransaction(operation, fruit, quantity);

```

You return a FruitTransaction even for invalid input (e.g., negative quantity, blank fruit). Ensure you validate operation code mapping, fruit and quantity before constructing FruitTransaction and throw informative RuntimeException on any invalid input. Also prefer RuntimeException types/messages consistent with the rest of the application per: "Handle and propagate errors via RuntimeException with informative messages; do not print stack traces or swallow exceptions."